business/sys/database/database.go

```diff
@@ -78,6 +78,15 @@ func Open(cfg Config) (*sqlx.DB, error) {
 // StatusCheck returns nil if it can successfully talk to the database. It
 // returns a non-nil error otherwise.
 func StatusCheck(ctx context.Context, db *sqlx.DB) error {
+	// If there is no timeout in parent context
+	// stop using hanging out on ping.
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		deadline = time.Now().Add(time.Second)
+	}
+	ctx, cancel := context.WithDeadline(ctx, deadline)
```

Comment:
Why not use content.WithTimeout and do everything in the if statement

```
if !ok {
   ctx, cancel = context.WithTimeout(ctx, time.Second)
   defer cancel()
}

Reply:
This syntax won't work

Reply:
```
if _, ok := ctx.Deadline(); !ok {
	var cancel context.CancelFunc
	ctx, cancel = context.WithTimeout(ctx, time.Second)
	defer cancel()
}
```

Reply:
That reduced the logic down to 4 lines and only touched the context if there is no deadline. What do you think?

Reply:
It looks better and shorter than my implementation. Sorry being late to respond.